mlir/test/python/ir/auto_location.py

```diff
@@ -0,0 +1,101 @@
+# RUN: %PYTHON %s | FileCheck %s
+# REQUIRES: python-ge-311
+import gc
+from contextlib import contextmanager
+
+from mlir.ir import *
+from mlir.dialects._ods_common import _cext
+from mlir.dialects import arith, _arith_ops_gen
+
+
+def run(f):
+    print("\nTEST:", f.__name__)
+    f()
+    gc.collect()
+    # assert Context._get_live_count() == 0
+
+
+@contextmanager
+def with_infer_location():
+    _cext.globals.set_loc_tracebacks_enabled(True)
+    yield
+    _cext.globals.set_loc_tracebacks_enabled(False)
+
+
+# CHECK-LABEL: TEST: testInferLocations
+@run
+def testInferLocations():
+    with Context() as ctx, Location.unknown(), with_infer_location():
+        ctx.allow_unregistered_dialects = True
+
+        op = Operation.create("custom.op1")
+        one = arith.constant(IndexType.get(), 1)
+        _cext.globals.register_traceback_file_exclusion(arith.__file__)
```

Comment:
ls there anyway to undo inclusions and exclusions? Not that I see that much value to it...

Reply:
After undoing this once per doing the opposite in-/ex-clusion, there's no way back though, right?

Reply:
yea that one is "stickier" IIRC (i exactly had some trouble flipping them on and off twice because of how the caches are setup) but i reasoned as you that it's not a very useful thing because files aren't usually generated dynamically in a single run (most people will just set these once on startup).